runtime/moonbase/src/lib.rs

```diff
@@ -1126,6 +1125,113 @@ where
 	}
 }
 
+// Name of the pallet to be removed in the migration
+// The pallet name is used as storage key prefix for all the values stored by pallets
+frame_support::parameter_types! {
+	pub const LocalAssetsPalletName: &'static str = "LocalAssets";
+}
+
+pub struct RemoveLocalAssets<R>(sp_std::marker::PhantomData<R>);
+
+impl<Runtime> Migration for RemoveLocalAssets<Runtime>
+where
+	Runtime: frame_system::Config,
+	Runtime::AccountId: Into<H160>,
+	Runtime: pallet_evm_precompileset_assets_erc20::AccountIdAssetIdConversion<
+		Runtime::AccountId,
+		AssetId,
+	>,
+	Runtime: pallet_evm::Config,
+{
+	fn friendly_name(&self) -> &str {
+		"MM_RemovePalletAssets"
+	}
+
+	fn migrate(&self, _available_weight: Weight) -> Weight {
+		log::info!("Removing (LocalAssets) pallet storage...");
+		let weight = frame_support::migrations::RemovePallet::<
+			LocalAssetsPalletName,
+			<Runtime as frame_system::Config>::DbWeight,
+		>::on_runtime_upgrade();
+
+		let addresses = precompiles::DisabledLocalAssets::<Runtime>::get();
+
+		for address in addresses.iter() {
+			pallet_evm::AccountCodes::<Runtime>::remove(address);
```

Comment:
We should not remove the dummy codes, it's useless because anyone can read them with the precompiles registry.
Also, if we remove the dummy code we can break solididy SCs because the solidity compiler add a pre-check before each call that esjure that the code is not empty

Reply:
When destroying any local asset, we remove the account code. This migration was just to make it consistent.

Reply:
It's another bug :/ 
We should never remove the dummy account code associated to a precompile

Reply:
I can remove it if you prefer, but those account codes will never be used again.

Reply:
They will be used by SC that call theses removed precompiles, so they will be used again, we should never remove dummy codes

Reply:
The migration is based on this: https://github.com/moonbeam-foundation/moonbeam/blob/c36c8ad555e9d1432013c0f8e97f997d9695d1c7/runtime/moonbase/src/asset_config.rs#L239

Reply:
I have removed this logic in commit https://github.com/moonbeam-foundation/moonbeam/pull/2634/commits/b47dd041f54fb9d6c812911c79f3ff883c99a1a0